### PR TITLE
Ants 2203 migrate bamboo to airbyte

### PR DESCRIPTION
**Pull request by NatElkins**
Updates Bamboo source for our use case.

### CODE REVIEW

File: airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/source.py [hunk 1]
```diff
@@ -26,146 +42,441 @@ def __init__(self, config: Mapping[str, Any]) -> None:
...
     def http_method(self) -> str:
         return "POST"
 
-    def request_body_json(self, **kwargs) -> Optional[Mapping]:
-        return {"title": "Airbyte", "fields": list(self.schema["properties"].keys())}
+    @staticmethod
+    def get_default_bamboo_fields() -> List[str]:
+        # As per https://documentation.bamboohr.com/docs/list-of-field-names
+        return [
+            "acaStatus"
```
Comment (junquanlim) [inline airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/source.py]: Let's remove this as it's deprecated

**Review by junquanlim — APPROVED** [review]

File: airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/source.py (hunk 1 above)
Comment (NatElkins) [reply airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/source.py]: My only concern is we may encounter some client with a legacy setup that is still making use of it (and not the new version, acaStatusCategory).  Seems like we might want to err on the side of caution and just try to pull it in.  We can choose to use it (or not) at the transform layer.  What do you think?

Comment (junquanlim) [reply airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/source.py]: should be fine, I think we just need to check that the deprecated fields don't result in weird bamboo endpoint error. I rmb previously when we included a deprecated field `flsaCode` it basically shifted all the values and keys and caused the values to be on the wrong keys in the response.

Comment (NatElkins) [reply airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/source.py]: Got it, I'll remove then.  We can always add it back if someone complains (unlikely).

**Review by junquanlim — APPROVED** [review]
LGTM!

**Pull request merged**